AndroidSDKTests/src/test/java/com/leanplum/internal/RequestTest.java

```diff
@@ -67,6 +68,20 @@ public void setUp() {
     Leanplum.setApplicationContext(context);
   }
 
+  /**
+   * Test that read writes happened sequentially when calling sendNow().
+   */
+  @Test
+  public void testSequenceOfDatabaseOperations() {
+    Map<String, Object> params = new HashMap<>();
+    params.put("data1", "value1");
+    params.put("data2", "value2");
+    ThreadWaiter waiter = new ThreadWaiter();
+    Request request = new Request("POST", Constants.Methods.START, params, waiter);
+    request.setAppId("fskadfshdbfa", "wee5w4waer422323");
+    request.sendIfConnected();
+    waiter.assertCallSequence();
```

Comment:
in order to make the test more real, we can have semaphores and control them in the test:

so before we call sendIfConnected, we allow the read, but block the write, then we call send(), and wait for 400ms, then we unblock the write, and then assert that the sequence is correct.